src/package/package.ts

```diff
@@ -258,6 +266,10 @@ export class Package {
         Object.entries(options).filter(([, value]) => value !== undefined)
       ) as PackageUpdateOptions;
 
+      if (opts.AppAnalyticsEnabled !== undefined && this.options.connection.getApiVersion() < '59.0') {
```

Comment:
@stephen-carter-at-sf have you considered just letting the update fail and deal with the error in the catch block, if you want provide a better message, that is.

Reply:
The suggestion to let it fail and catch it goes against my instinctive defensive programming mentality. I've found its best to assert state before attempting any code that has mutation involved. That is, I wouldn't want to perform an update and then let it fail because it trusts the update function too much in my opinion. Also my guess is that the catch block would catch generic exception types that would require analysis to map it back to the correct error message. So I think I like this assertion before update better.